ONSdigital/terraform-aws-events
# Fixes #51 Make CI principals configurable.

## What is the context of this PR?
DP have a requirement to be able to override the default CI user/role with their own principals. Currently these values are hardcoded into the module. This PR introduces a new variable to allow overriding the `ci_principals` which accepts a list of string values. Terraform's [coalescelist](https://www.terraform.io/language/functions/coalescelist) built-in function is then used to default to the module's standard `user/ci` and `role/ci` principals if the principals are not supplied through the variable.

## How to test?
All existing CI checks/tests should continue to pass. This change will be integrated with `dp-setup` project and tested in the `dp-sandbox` environment.
